kolibri/core/assets/src/vue/core-modal/index.vue

```diff
@@ -109,6 +102,9 @@
       closeModal() {
         this.$dispatch('close');
       },
+      submitModal() {
+        this.$dispatch('submit');
+      },
```

Comment:
Rereading your message, it looks like I misunderstood what you had done with your events - they weren't `submit` or `escape` events per se - those were just names given to specific interaction (keyboard/mouse) events that might make sense in modal contexts. 

If that were the case, wouldn't you have to implement things like `close` and `submit` with every listener/parent component?